apps/test/unit/templates/curriculumCatalog/curriculumCatalogExpandedCardTest.jsx

```diff
@@ -0,0 +1,256 @@
+    const professional_learning_program = 'https://code.org/apply';
+    const self_paced_pl_course_offering_path = '/courses/vpl-csa-2023';
+    renderCurriculumExpandedCard({
+      ...defaultProps,
+      professionalLearningProgram: professional_learning_program,
+      selfPacedPlCourseOfferingPath: self_paced_pl_course_offering_path,
+    });
+
+    screen.getByText('Professional Learning');
+    screen.getByText('Facilitator led workshops');
+    screen.getByText('Self-paced PL');
+  });
+
+  it('does not render professional learning section when no professional learning available', () => {
+    renderCurriculumExpandedCard();
+
+    expect(screen.queryByText('Profession Learning')).to.be.null;
+  });
+
+  it('does not render professional learning section when not in US', () => {
+    const professional_learning_program = 'https://code.org/apply';
+    const self_paced_pl_course_offering_path = '/courses/vpl-csa-2023';
+    renderCurriculumExpandedCard({
+      ...defaultProps,
+      professionalLearningProgram: professional_learning_program,
+      selfPacedPlCourseOfferingPath: self_paced_pl_course_offering_path,
+      isInUS: false,
+    });
+
+    expect(screen.queryByText('Profession Learning')).to.be.null;
+  });
+
+  it('renders devices and their correct icons', () => {
+    const deviceIcons = {
+      ideal: 'circle-check',
+      not_recommended: 'triangle-exclamation',
+      incompatible: 'circle-xmark',
+    };
+    renderCurriculumExpandedCard();
+
+    const devices = JSON.parse(defaultProps.deviceCompatibility);
+
+    Object.keys(devices).forEach(device => {
+      const newDevice = device === 'no_device' ? 'Offline' : device;
+      const deviceElement = screen.getByText(newDevice, {exact: false});
+
+      const parentElement = deviceElement.parentElement;
+      expect(
+        parentElement.querySelectorAll(
+          `i[class*=${deviceIcons[devices[device]]}]`
+        )
+      ).to.have.length(1);
+    });
+  });
+
+  it('clicking assign button triggers onAssign function', () => {
+    const onAssign = sinon.spy();
+    renderCurriculumExpandedCard({
+      ...defaultProps,
+      assignButtonOnClick: onAssign,
+    });
+
+    expect(onAssign).not.to.have.been.called;
+
+    const assignButton = screen.getByRole('button', {
+      name: new RegExp(
+        `Assign ${defaultProps.courseDisplayName} to your classroom`
+      ),
+    });
+
+    fireEvent.click(assignButton);
+
+    expect(onAssign).to.have.been.calledOnce;
+  });
+
+  it('clicking close button triggers onClose function', () => {
+    const onClose = sinon.spy();
+    const {container} = renderCurriculumExpandedCard({
+      ...defaultProps,
+      onClose: onClose,
+    });
+
+    expect(onClose).not.to.have.been.called;
+
+    const [closeIcon] = container.querySelectorAll('i[class*=fa-xmark]');
+    const onCloseButton = closeIcon.parentElement.parentElement;
```

Comment:
Instead of these lines and setting the `renderCurriculumExpandedCard` to `{container}`, could you do what you did in the test above with `screen.getByRole('button', {...})`?